### ISSUE TITLE
[BUG]: "Mark as watched" does not work

### ISSUE THREAD

**Issue by Norsagir** [yuliskov/SmartTube#4177]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/yuliskov/SmartTube/issues) or [closed](https://github.com/yuliskov/SmartTube/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://github.com/yuliskov/SmartTube#faq) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.

### Affected version

24.89

### Device Type

Smart TV/Box

### Affected Android

Shield TV 2019 Pro

### Steps to reproduce the bug

1. Go to the Subscription panel
2. Long press on a video to show menu
3. Click "Mark as watched"
4. Go to another menu like "Home", then go back to Subscription

### Actual behavior

The video is not marked as watched.

**Comment by TheColdOne** [comment]
I've noticed that it DOES work on the backend. Like, if you check your play history, it should be there at the top now. Then if you go to the official Youtube app, the video will be marked as watched.

There seems to be a bigger issue recently with the app not getting that updated info on the backend to show the state on the thumbnail, though I believe everything is still working as it should concerning history, recommendations, etc.

At least this has been my observation for the last week or two.

**Comment by Norsagir** [comment]
Oh maybe you are right, I didn't notice it but I could do some experience.

But the main problem for me is that the app does not understand that the video is marked as watched, so no red bar but the main issue is that the 'hide watched videos' option is not working...

**Comment by TheColdOne** [comment]
Oh yeah, it would have that ripple affect. If the app itself doesn't see it as watched (red progress bar) then other functions like "hide watched" (or anything with watched videos) would also not function, regardless of Youtube server backend state.

**Comment by Norsagir** [comment]
Yeah, I think that just one thing is broken and it has impact on many features.

**Comment by mwlandof** [comment]
It's like that for a longer time (at least 2 updates already)... i do really wish to see it being fixed...

**Comment by Norsagir** [comment]
Me too, but as I can see, this is not a priority for the devs...

**Comment by TheColdOne** [comment]
For now, I've just started opening the videos and manually scrubbing to the end to get it properly marked off in the UI.

**Comment by mwlandof** [comment]
> For now, I've just started opening the videos and manually scrubbing to the end to get it properly marked off in the UI.

I do the same for a long time now…but it’s pain in the back

**Comment by BimbleBobSki** [comment]
> For now, I've just started opening the videos and manually scrubbing to the end to get it properly marked off in the UI.

I have found if you scrub right to the end it doesn't always mark it as watched, so I scrub through to almost the end, but let the last few seconds play until the end and then it marks it as watched OK.

**Comment by StormPooper** [comment]
Previously I found that the video would be saved to my history in YouTube as watched and given the red marked indicator in the UI, but the UI would later not show it as watched once reloaded (restarting the app, viewing it in another section, etc), but with the recent builds resolving the various 403 issues (27.03 stable) it now appears to also not show the video in the history section, although it does actually save it to YouTube. In the latest beta, 27.16, it is back to the old behaviour of appearing in the history but forgetting the watched flag in the UI. Interestingly, any statuses it syncs from YouTube that were done via another instance of SmartTube is marked correctly in the UI, so videos marked as watched in the beta appear watched in the stable version and vice versa.

**Comment by Hny-B-King** [comment]
It’s still not working in latest stable build

**Comment by BimbleBobSki** [comment]
> It’s still not working in latest stable build

I am now with the latest stable version getting videos showing as partially watched even though they have been fully watched as well as the existing issue.

The marking read works when you first do it, but once you leave that list/subscription and go back it it not marked.

**Comment by StormPooper** [comment]
On the latest stable build I now get a partially watched video when marking as watched, but it does show the progress amount on all views.

**Comment by MrSourceUnknown** [comment]
It's unfortunate that this is wrongly listed as 'fixed' in both the Github release notes (see 30.16 stable, https://github.com/yuliskov/SmartTube/releases/tag/30.16s) and also in the most recent in-app changelog (30.43).
Yet the underlying issue remains.

I hope this issue is still actively looked into, because "Mark as Watched" is one of the controls I use most often.

**Comment by rafaelenescom** [comment]
I'm still having inconsistent behavior when watching videos until the end and randomly not being marked as watched.

**Comment by BimbleBobSki** [comment]
> I'm still having inconsistent behavior when watching videos until the end and randomly not being marked as watched.

Same here on both my Humax Aura Freeview recorder and my FireStick.  

As you say, totally random, some programs show as fully watched others show as just watched, but when you play them again they start from the beginning, so they are being marked as watched behind the scenes, but the red line is not doing what it should.
